pkgs/servers/jellyfin/server-generic.nix

```diff
@@ -0,0 +1,216 @@
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+
+    export DOTNET_CLI_TELEMETRY_OPTOUT=1
+    export DOTNET_NOLOGO=1
+
+    nuget sources Add -Name nixos -Source "$PWD/nixos"
+    nuget init "$nugetDeps" "$PWD/nixos"
+
+    # FIXME: https://github.com/NuGet/Home/issues/4413
+    mkdir -p $HOME/.nuget/NuGet
+    cp $HOME/.config/NuGet/NuGet.Config $HOME/.nuget/NuGet
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    dotnet publish Jellyfin.Server \
+      --configuration Release \
+      --runtime ${runtimeId} \
+      --no-self-contained \
+      --output $out/opt/jellyfin
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    makeWrapper ${dotnet-aspnetcore}/bin/dotnet $out/bin/jellyfin \
+      --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDeps}" \
+      --add-flags "$out/opt/jellyfin/jellyfin.dll" \
+      --add-flags "--ffmpeg ${ffmpeg}/bin/ffmpeg" \
+      --add-flags "--webdir ${jellyfin-web}/share/jellyfin-web"
+
+    runHook postInstall
+  '';
+
+  passthru.tests = {
+    smoke-test = nixosTests.jellyfin;
+  };
+
+  passthru.updateScript = writeScript "${pname}-update-script" ''
+    #!${stdenv.shell}
+    PATH=${lib.makeBinPath [ curl jq common-updater-scripts dotnet-sdk gnused nix coreutils findutils ]}
+    set -eo pipefail
+
+    deps_file="${toString nugetDepsFile}"
+
+    latestVersion="$(curl -s "https://api.github.com/repos/jellyfin/jellyfin/releases" | jq -r '${limitVersionRangeSelector}.[0].tag_name' | sed 's/^v//')"
+    if [[ "${version}" == "$latestVersion" ]]; then
+      echo "${updateAttr} is up-to-date: ${version}"
+      exit 0
+    fi
+
+    update-source-version ${updateAttr} 0 0000000000000000000000000000000000000000000000000000000000000000 --file=${versionFile} --version-key=version
+    update-source-version ${updateAttr} $latestVersion --file=${versionFile} --version-key=version
+
+    store_src="$(nix-build . -A ${updateAttr}.src --no-out-link)"
+    src="$(mktemp -d /tmp/jellyfin-src.XXX)"
+    echo "Temp src dir: $src"
+    cp -rT "$store_src" "$src"
+    chmod -R +w "$src"
+
+    pushd "$src"
+
+    mkdir ./nuget_tmp.packages
+    dotnet restore Jellyfin.Server --packages ./nuget_tmp.packages --runtime linux-x64
+
+    echo "{ fetchNuGet }: [" >"$deps_file"
+    while read pkg_spec; do
+      { read pkg_name; read pkg_version; } < <(
+        # Build version part should be ignored: `3.0.0-beta2.20059.3+77df2220` -> `3.0.0-beta2.20059.3`
+        sed -nE 's/.*<id>([^<]*).*/\1/p; s/.*<version>([^<+]*).*/\1/p' "$pkg_spec")
+      pkg_sha256="$(nix-hash --type sha256 --flat --base32 "$(dirname "$pkg_spec")"/*.nupkg)"
+      cat >>"$deps_file" <<EOF
+      (fetchNuGet {
+        name = "$pkg_name";
+        version = "$pkg_version";
+        sha256 = "$pkg_sha256";
+      })
+    EOF
+    done < <(find ./nuget_tmp.packages -name '*.nuspec' | sort)
+    echo "]" >>"$deps_file"
+
+    popd
+    rm -r "$src"
```

Comment:
Can we move this into a shell file to get proper highlighting, formatting and easier shellcheck support?

Reply:
I've moved the update logic to separate shell scripts. Also fixed some warnings from `shellcheck`.

I've added some arguments for the needed variables, which i pass from the `passthru.updateScript` to the shell scripts.